L1Trigger/TrackFindingTracklet/src/TrackletEventProcessor.cc

```diff
@@ -14,16 +14,16 @@
 using namespace trklet;
 using namespace std;
 
-TrackletEventProcessor::TrackletEventProcessor() = default;
+TrackletEventProcessor::TrackletEventProcessor() : settings_(nullptr) {}
 
 TrackletEventProcessor::~TrackletEventProcessor() {
-  if (settings_->bookHistos()) {
+  if (settings_ && settings_->bookHistos()) {
     histbase_->close();
   }
 }
 
-void TrackletEventProcessor::init(const Settings* theSettings) {
-  settings_ = theSettings;
+void TrackletEventProcessor::init(Settings const& theSettings) {
+  settings_ = &theSettings;
```

Comment:
the issue is not created by this PR - however there is no promise that the memory allocated to theSettings will stay around for the lifetime of TrackletEventProcessor (which this construct depends on)